### PR TITLE
Fix version 1 exams missing counterInExercise field

### LINKED ISSUE #4940: make sure unit test examples include all exam versions
### Observed behavior

Our sample server responses only include v1 exams

### Expected behavior

we should have all versions represented

### Context

0.12

### LINKED ISSUE #5427: 'NaN' in titles of difficult questions
### Observed behavior

image

image

### Expected behavior

When there is more than one question drawn from a single exercise, the questions should be numbered with incrementing integers

### User-facing consequences

Confusion

### Errors and logs

None

### Steps to reproduce

See it live on the demo server: http://kolibridemo.learningequality.org/coach/#/f6e7f562bb0ea654340c726e1f81bbe9/reports/quizzes/448067d8b82df0a03902ea2aeff939e2/questions

### Context

0.12.2

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Adds a conversion function that adds a missing field to some Version 1 Exams
1.  Combines conversion functions into one that handles V0 -> v2 and V1 -> V2 conversion.
1. Updates tests with examples from both V0 and V1 (Fixes #4940 )

### Reviewer guidance

1. Make some exams ~using the current 0.12 branch, which should create exams that are V1~. Edit the Exam models in the shell, or via a REST tool to create Exams whose `question_sources` items are missing the `counterInExercise` field and also have a `data_model_version` field equal 0. Confirm that the problem in #5427 occurs
1. Update to this fork and verify that it is fixed.

### References

Fixes #5427

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/exams/models.py [hunk 1]
```diff
@@ -32,7 +32,19 @@ class Exam(AbstractFacilityDataModel):
...
+    The `question_sources` field contains different values depending on the 'data_model_version' field.
+
+    V2:
+        Similar to V1, but with a `counterInExercise` field
+        [
+            {
+                "exercise_id": <exercise_pk>,
+                "question_id": <item_id_within_exercise>,
+                "title": <exercise_title>,
+                "counterInExercise": <unique_count_for_question>
```
Comment (indirectlylit) [inline kolibri/core/exams/models.py]: should we make this `counter_in_exercise` for consistency?

Comment (jonboiser) [reply kolibri/core/exams/models.py]: Yeah, let's do that. So to convert a V1 exam with `counterInExercise`,we'll just rename it to `counter_in_exercise`.

**Comment by indirectlylit** [comment]
my guess/hope is that this will also address #5425

**Comment by nucleogenesis** [comment]
@jonboiser 

I tested the latest commits and they resolved the issues we discussed on Slack. When there are no `counter_in_exercise` fields then the numbers are displayed as expected.

There appears to be a failing test (which I get locally as well running `yarn test`): https://travis-ci.org/learningequality/kolibri/jobs/526240872 - once that's resolved then this should be ready to roll.

**Review by nucleogenesis — APPROVED** [review]
@jonboiser Looks great!

**Pull request merged**